### PR TITLE
Bug 1145473 - [email/backend] MailAPI's bridge-processing logic may break.

### PR DESCRIPTION
**Pull request by mcav**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=try-taskcluster&revision=1ef50ca1a262701385a50db480832cd6b7a2ac31

File: js/mailapi.js [hunk 1]
```diff
@@ -2070,7 +2069,8 @@ function reportError() {
     else
       msg = "" + arguments[i];
   }
-  throw new Error(msg);
+  // When in tests, this will fail the test; when not in tests, we just log.
+  logic.fail(new Error(msg));
```
Comment (asutherland) [inline js/mailapi.js]: Although logic should be universally available, you didn't require() it AFAICT.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=try-taskcluster&revision=ef99ff2ea0a1776762c7de31cd77d7f5b14e5d54

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=try-taskcluster&revision=daac4b94f32e77524407999122d749effa24e5fb

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=try-taskcluster&revision=49223adefb6057d1be1985a1fdf82575f3930373

**Pull request merged**